docs/src/main/asciidoc/secret-management.adoc

```diff
@@ -116,26 +120,55 @@ spring:
 @SpringBootApplication
 public class SampleApplication implements CommandLineRunner {
 
-    @Value("${secret-name-in-key-vault-1}")
-    private String secretNameInKeyVault1;
-    @Value("${secret-name-in-key-vault-2}")
-    private String secretNameInKeyVault2;
-    @Value("${secret-name-in-key-vault-both}")
-    private String secretNameInKeyVaultBoth;
+    @Value("${sampleProperty1}")
+    private String sampleProperty1;
+    @Value("${sampleProperty2}")
+    private String sampleProperty2;
+    @Value("${samplePropertyInMultipleKeyVault}")
+    private String samplePropertyInMultipleKeyVault;
 
     public static void main(String[] args) {
         SpringApplication.run(SampleApplication.class, args);
     }
 
     public void run(String[] args) {
-        System.out.println("secretNameInKeyVault1: " + secretNameInKeyVault1);
-        System.out.println("secretNameInKeyVault2: " + secretNameInKeyVault2);
-        System.out.println("secretNameInKeyVaultBoth: " + secretNameInKeyVaultBoth);
+        System.out.println("sampleProperty1: " + sampleProperty1);
+        System.out.println("sampleProperty2: " + sampleProperty2);
+        System.out.println("samplePropertyInMultipleKeyVault: " + samplePropertyInMultipleKeyVault);
     }
 
 }
 ----
 
+=== Advanced Usage
+
+==== Special characters in property name
+Key Vault secret name only support characters in `[0-9a-zA-Z-]`. Refs: link:https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name[Vault-name and Object-name]. If your property name contains other characters, you can use these workarounds:
+
+===== Use `-` instead of `.` in secret name
+
+`.` is not supported in secret name. If your application have property name which contain `.`, like `spring.datasource.url`, just replace `.` to `-` when save secret in Azure Key Vault. For example: Save `spring-datasource-url` in Azure Key Vault. In your application, you can still use `spring.datasource.url` to retrieve property value.
+
+===== Use property placeholders
+
+For example: setting this property in your application.properties:
+[source,properties]
+----
+property.with.special.character__=${propertyWithoutSpecialCharacter}
+----
+
+The application will get  `propertyWithoutSpecialCharacter` key name and assign its value to `property.with.special.character__`.
+
+==== Case Sensitive
+
+To enable case-sensitive mode, you can set the following property:
+
+[source,properties]
+----
+spring.cloud.azure.keyvault.secret.property-sources[].case-sensitive=true
+----
+
+
 === Samples
 
 Please refer to link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0[azure-spring-boot-samples] for more details.
```

Comment:
I think it's better to link the keyvault samples here.

Reply:
Updated.